src/auth/Scheme.cc

```diff
@@ -61,10 +61,15 @@ Auth::Scheme::FreeAll()
 {
     assert(shutting_down);
 
+    if (!_Schemes)
+        return;
+
     while (GetSchemes().size()) {
         Auth::Scheme::Pointer scheme = GetSchemes().back();
         GetSchemes().pop_back();
         scheme->shutdownCleanup();
     }
+    delete _Schemes;
```

Comment:
I am not excited about this change because it goes against our [NoNewGlobals](https://wiki.squid-cache.org/DeveloperResources/SquidCodingGuidelines#rule-no-new-globals) pattern (that this code has not been migrated to) and increases code exposure to the dangerous _Schemes variable (instead of removing it like NoNewGlobals would), albeit in a trivial way. 

I will not stand in the way if others like it though.

Reply:
Dealt with